### PR TITLE
Tests for Metric Server

### PR DESCRIPTION
**Pull request by pdewilde**

### CODE REVIEW

File: cmd/main_test.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+func marshalRequest(tb testing.TB, req *metrics.SendMetricRequest) io.Reader {
+	tb.Helper()
+	b, err := json.Marshal(req)
+	if err != nil {
+		tb.Fatalf("could not marshal json: %s", err.Error())
+	}
+	return bytes.NewReader(b)
+}
+
+func Test_handleMetric(t *testing.T) {
```
Comment (drevell) [inline cmd/main_test.go:70]: The name `TestHandleMetric` would be conventional here

File: cmd/main_test.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		name       string
+		db         server.MetricsLookuper
+		body       io.Reader
+		wantStatus int
+		wantLogs   map[*slogassert.LogMessageMatch]int
+	}{
+		// TODO: more test cases if by some miracle this ugly thing doesn't get ousted in code review
+		{
+			name: "happy_single_metric",
+			db: &testMetricsDB{apps: map[string]*server.AppMetrics{"test": {
```
Comment (drevell) [inline cmd/main_test.go:82]: Typically you'd break this up over multiple lines. 3 open braces on the same line is perhaps a bit much

File: cmd/main_test.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+			if got, want := response.StatusCode, tc.wantStatus; got != want {
+				t.Errorf("unexpected response code. got %d want %d", got, want)
+			}
+
+			// Normally we wouldn't test log messages, but as that is the way metrics
+			// are being exported, it seems important to do so here.
+			for k, v := range tc.wantLogs {
+				// TODO: I don't like that this panics if there are no matches, would rather handle error myself
+				// I have https://github.com/thejerf/slogassert/pull/5 to try and fix it.
+				if got, want := logHandler.AssertSomePrecise(*k), v; got != want {
```
Comment (drevell) [inline cmd/main_test.go:139]: I don't think you need a new variable `want`, you can just rename `v` to `want`. They always have the same value.

File: cmd/main_test.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+			defer response.Body.Close()
+
+			if got, want := response.StatusCode, tc.wantStatus; got != want {
+				t.Errorf("unexpected response code. got %d want %d", got, want)
+			}
+
+			// Normally we wouldn't test log messages, but as that is the way metrics
+			// are being exported, it seems important to do so here.
+			for k, v := range tc.wantLogs {
+				// TODO: I don't like that this panics if there are no matches, would rather handle error myself
```
Comment (drevell) [inline cmd/main_test.go:137]: The panic does seem to violate https://google.github.io/styleguide/go/decisions#assertion-libraries, but I don't think it's a dealbreaker.

Also, I think you can use `logHandler.Assert()` instead, which doesn't panic.

**Review by drevell — APPROVED** [review]

File: cmd/main_test.go (hunk 4 above)
Comment (pdewilde) [reply cmd/main_test.go:137]: I want to use `logHandler.Assert()` but I want the logic given by the "Precise" match (looks at more than just message), but to do so I need to export the function that allows it. 

I have a pr https://github.com/thejerf/slogassert/pull/5 so if that gets merged I will be able to switch to using `logHandler.Assert()`

File: cmd/main_test.go (hunk 2 above)
Comment (pdewilde) [reply cmd/main_test.go:82]: Broken up. If only golang had sets that would have a super clean oneliner

File: cmd/main_test.go (hunk 4 above)
Comment (pdewilde) [reply cmd/main_test.go:140]: Changed TODO to make it more clear my plans

**Pull request merged**